libmscore/hairpin.cpp

```diff
@@ -173,13 +184,20 @@ void HairpinSegment::draw(QPainter* painter) const
       else
             color = hairpin()->curColor();
       QPen pen(color, point(hairpin()->lineWidth()), hairpin()->lineStyle());
+      if (hairpin()->lineStyle() == Qt::CustomDashLine) {
+            QVector<qreal> pattern;
+            pattern << 5.0 << 20.0;
```

Comment:
These numbers are in units of whatever the line width is.  That is, if the line width is 0.10sp, and that's the unit applied to the 5 & 20.  So this code sets the pattern to 0.5sp dashes with 2sp gaps.  Meaning it automatically scales with sp as well as with the line width.  I found it odd that Qt has you specify it this way, but in practice, when I tried it out, it does scale well.